src/lib/image/create/create.c

```diff
@@ -57,20 +53,42 @@ int _singularity_image_create(struct image_object *image, long int size) {
     fprintf(image_fp, LAUNCH_STRING); // Flawfinder: ignore (LAUNCH_STRING is a constant)
 
     singularity_message(VERBOSE2, "Growing image to %ldMB\n", size);
-    // TODO: there are likely better ways to do this (falloc?); further, we should really handle
-    // EINTR here.
-    for(i = 0; i < size; i++ ) {
-        if ( fwrite(buff, 1, BUFFER_SIZE, image_fp) < BUFFER_SIZE ) {
-            singularity_message(ERROR, "Failed allocating space to image: %s\n", strerror(errno));
-            ABORT(255);
+    while(1)
+    {
+        retval = posix_fallocate(image->fd, sizeof(LAUNCH_STRING), size*BUFFER_SIZE);
```

Comment:
As far as I understand, it's filled with zero. A quick test seems to confirm this.